tensornetwork/tests/backend_contextmanager_test.py

```diff
@@ -0,0 +1,53 @@
+import tensornetwork as tn
+import pytest
+import numpy as np
+
+def test_contextmanager_simple():
+  tn.set_default_backend("pytorch")
+  with tn.DefaultBackend("numpy"):
+    a = tn.Node(np.ones((10,)))
+    b = tn.Node(np.ones((10,)))
+  assert a.backend.name == b.backend.name
+  edge = a[0] ^ b[0]
+  final_node = tn.contract(edge)
+  assert final_node.tensor == 10.0
+
+
+def test_contextmanager_interaption():
+  tn.set_default_backend("pytorch")
+  assert tn.config.default_backend == "pytorch"
+  with tn.DefaultBackend("numpy"):
+    assert tn.config.default_backend == "pytorch"
+    a = tn.Node(np.ones((10,)))
+    tn.set_default_backend("tensorflow")
```

Comment:
using `set_default_backend` while there is a backend stack feels like a foot gun. Can you raise an error in this case instead?

Reply:
Sorry I didn't get what you mean by raising an error in this case. Prohibit using `set_default_backend` while in `with` statement?